compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -17752,30 +17752,36 @@ TR::Node *fmaxminSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
    TR::Node * secondChild = node->getSecondChild();
    bool isBothConst = firstChild->getOpCode().isLoadConst() && secondChild->getOpCode().isLoadConst();
    float fmin = 0, fmax = 0;
-   bool maxOpcode = node->getOpCodeValue() == TR::fmax;
+   bool isMaxOpcode = node->getOpCodeValue() == TR::fmax;
 
-   if (isBothConst)
+   if (!isBothConst) return node;
+
+   float first = firstChild->getFloat();
+   float second = secondChild->getFloat();
+
+   uint32_t firstBits = firstChild->getFloatBits();
+   uint32_t secondBits = secondChild->getFloatBits();
+
+   if (isNaNFloat(firstChild))
```

Comment:
May I ask you to add a comment here (and for the `dmaxminSimplifier`) explicitly stating that if either operand is NaN, the result is the NaN operand, and that -0.0f is treated as less than 0.0f (and -0.0 is treated as less than 0.0, respectively), just in case it's not immediately clear?

Reply:
can you review again please @hzongaro